trio/tests/test_socket.py

```diff
@@ -990,3 +990,26 @@ async def receiver():
             nursery.start_soon(receiver)
             await wait_all_tasks_blocked()
             a.close()
+
+
+async def test_many_sockets():
+    total = 5000  # Must be more than MAX_AFD_GROUP_SIZE
+    sockets = []
+    for x in range(total // 2):
+        try:
+            a, b = stdlib_socket.socketpair()
+        except OSError as e:
+            if e.errno != 24:
+                raise  # pragma: no cover
+            else:
+                break
+        sockets += [a, b]
+    async with _core.open_nursery() as nursery:
+        for s in sockets:
+            nursery.start_soon(_core.wait_readable, s)
+        await _core.wait_all_tasks_blocked()
+        nursery.cancel_scope.cancel()
+    for sock in sockets:
+        sock.close()
+    if x != total // 2 - 1:
+        pytest.xfail(f"Unable to open more than {(x-1)*2} sockets.")
```

Comment:
Let's not use `xfail` here -- to me `xfail` means "there's a known bug in the package", and this isn't a bug in trio at all, it's just a limitation of the test environment. We could do a `print` just for info and debugging?